homeassistant/components/smappee.py

```
@@ -0,0 +1,290 @@
+        #   datetime and pandas timestamps
+        # Aggregation:
+        # 1 = 5 min values (only available for the last 14 days),
+        # 2 = hourly values,
+        # 3 = daily values,
+        # 4 = monthly values,
+        # 5 = quarterly values
+        if self.is_remote_active:
+            end = datetime.utcnow()
+            start = end - timedelta(minutes=delta)
+            try:
+                return self._smappy.get_consumption(location_id,
+                                                    start,
+                                                    end,
+                                                    aggregation)
+            except RequestException as error:
+                _LOGGER.error(
+                    "Error getting comsumption from Smappee cloud. (%s)",
+                    error)
+                return False
+        return False
+
+    def get_sensor_consumption(self, location_id, sensor_id):
+        """Update data from Smappee."""
+        # Start & End accept epoch (in milliseconds),
+        #   datetime and pandas timestamps
+        # Aggregation:
+        # 1 = 5 min values (only available for the last 14 days),
+        # 2 = hourly values,
+        # 3 = daily values,
+        # 4 = monthly values,
+        # 5 = quarterly values
+        if self.is_remote_active:
+            start = datetime.utcnow() - timedelta(minutes=30)
+            end = datetime.utcnow()
+            try:
+                return self._smappy.get_sensor_consumption(location_id,
+                                                           sensor_id,
+                                                           start,
+                                                           end, 1)
+            except RequestException as error:
+                _LOGGER.error(
+                    "Error getting comsumption from Smappee cloud. (%s)",
+                    error)
+                return False
+
+        return False
```

These error values are never handled. Probably better to propagate the error to the update method of the sensor.